### DISCUSSION TITLE
Getting data from api and transform them into the suitable form but still show value is null error.

### DISCUSSION THREAD

**Discussion by a1235456** [tradingview/lightweight-charts#1216] [Q&A] [answered]
I have got small amount of data to try using lightweight charts. The data are in the following shape, which I then further process them to fit the series data form.
![image](https://user-images.githubusercontent.com/24928108/206265414-dad6a881-2897-49ed-bd5d-4f9a117dd6b0.png)
However, after the transformation, the value is null error occurred.
![image](https://user-images.githubusercontent.com/24928108/206265163-adb81ccc-bc5d-4029-823c-e10cbde2f61f.png)

And something stranger happened, I have tried to use another set of data which looks like this.
![image](https://user-images.githubusercontent.com/24928108/206265843-b798d1fd-7a03-4c9d-b489-067ed4109dae.png)
The difference between the two data sets is just the timestamp, I suppose.

But this time, the graph shows normally.
![image](https://user-images.githubusercontent.com/24928108/206266072-9d186993-5f83-4abb-b404-b4cbf3f8db4b.png)

I am struggling to find the reason. Please, can someone help me out? Thank you so much.

**Comment by edew** [comment, marked as answer]
It looks like the times in the first screenshot might not be in ascending order?

If you share all the error logs from the console we might be able to give a better answer.

**Reply by a1235456** [reply]
Sorry that I forgot to reply. The api gave me the data in descending order.

**Comment by appsunited** [comment]
Your timestamp is not working, because JavaScript is in millseconds. Also had this issue before. To get it working you need to add three more trailing zeros to the timestamp. Instead of `"time":1670436420` add this `"time":1670436420000`
